watchtower/wtclient/backup_task.go

```diff
@@ -167,6 +199,10 @@ func (t *backupTask) bindSession(session *wtdb.ClientSessionBody) error {
 		weightEstimate.AddP2WKHOutput()
 	}
 
+	if t.chanType.HasAnchors() != session.Policy.IsAnchorChannel() {
+		panic("invalid session for task")
```

Comment:
Alternatively, could use the `Criticalf` logging here so things wind down a bit more gracefully. Not sure if the full goroutine/stack trace would be helpful here if this actually does happen in the wild for w/e reason.

Reply:
done